components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -373,6 +382,57 @@ public static AuthenticationContext getContextData(HttpServletRequest request) {
         return context;
     }
 
+    /**
+     * Retrieve authentication context data. If the AUTHENTICATION_CONTEXT_EXPIRY_VALIDATION is enabled and the cache
+     * entry is expired, an exception is thrown.
+     * If the AUTHENTICATION_CONTEXT_EXPIRY_VALIDATION is disabled, it will not validate the authentication context
+     * cache for session expiry.
+     *
+     * @param request Authentication request.
+     * @return Authentication context.
+     * @throws FrameworkClientException Authentication session has expired.
+     */
+    public static AuthenticationContext getContextDataWithExpiry(HttpServletRequest request)
```

Comment:
to avoid any code duplication with the deprecated method, we can make the involvement of context expiry passable with a boolean param